simtools/configuration/commandline_parser.py

```diff
@@ -612,18 +612,20 @@ def azimuth_angle(angle):
                 "The azimuth angle provided is not a valid astropy.Quantity. "
                 "Will check if it is (north, south, east, west) instead"
             )
-        azimuth_map = {
-            "north": 0 * u.deg,
-            "south": 180 * u.deg,
-            "east": 90 * u.deg,
-            "west": 270 * u.deg,
-        }
-        azimuth_angle = angle.lower()
-        if azimuth_angle in azimuth_map:
-            return azimuth_map[azimuth_angle]
-        raise argparse.ArgumentTypeError(
-            "The azimuth angle can only be one of " f"(north, south, east, west), not {angle}"
-        )
+        if isinstance(angle, str):
+            azimuth_map = {
+                "north": 0 * u.deg,
+                "south": 180 * u.deg,
+                "east": 90 * u.deg,
+                "west": 270 * u.deg,
+            }
+            azimuth_angle = angle.lower()
+            if azimuth_angle in azimuth_map:
+                return azimuth_map[azimuth_angle]
+            raise argparse.ArgumentTypeError(
+                "The azimuth angle can only be one of " f"(north, south, east, west), not {angle}"
```

Comment:
Is this clear here? We allow strings and floats in this function, the last check here is separate from above. Maybe better to have a seperate function or Change the message to something as "If the angle is pased as a string only ... is allowed and not ..".

Reply:
Good point - I've improved the error message.